src/other/mlpmixer.jl

```diff
@@ -87,3 +110,185 @@ end
+
+Creates a block for the ResMixer architecture.
+([reference](https://arxiv.org/abs/2105.03404)).
+
+# Arguments
+- `planes`: the number of planes in the block
+- `npatches`: the number of patches of the input
+- `mlp_ratio`: ratio of the number of hidden channels in the channel mixing MLP to the number 
+               of planes in the block
+- `mlp_layer`: the MLP block to use
+- `dropout`: the dropout rate to use in the MLP blocks
+- `drop_path_rate`: Stochastic depth rate
+- `activation`: the activation function to use in the MLP blocks
+- `λ`: initialisation constant for the LayerScale
+"""
+function resmixerblock(planes, npatches; mlp_ratio = 4.0, mlp_layer = mlp_block, 
+                       dropout = 0., drop_path_rate = 0., activation = gelu, λ = 1e-4)
+return Chain(SkipConnection(Chain(Flux.Diagonal(planes),
+                                  x -> permutedims(x, (2, 1, 3)),
+                                  Dense(npatches, npatches),
+                                  x -> permutedims(x, (2, 1, 3)),
+                                  LayerScale(planes, λ),
+                                  DropPath(drop_path_rate)), +),
+             SkipConnection(Chain(Flux.Diagonal(planes),
+                                  mlp_layer(planes, Int(mlp_ratio * planes); dropout, activation),
+                                  LayerScale(planes, λ),
+                                  DropPath(drop_path_rate)), +))
+end
+
+struct ResMLP
+  layers
+end
+
+"""
+    ResMLP(size::Symbol = :base; patch_size::Int = 16, imsize::NTuple{2} = (224, 224),
+           drop_path_rate = 0., nclasses = 1000)
+
+Creates a model with the ResMLP architecture.
+([reference](https://arxiv.org/abs/2105.03404)).
+
+# Arguments
+- `size`: the size of the model - one of `small`, `base`, `large` or `huge`
+- `patch_size`: the size of the patches
+- `imsize`: the size of the input image
+- `drop_path_rate`: Stochastic depth rate
+- `nclasses`: number of output classes
+
+See also [`Metalhead.mlpmixer`](#).
+"""
+function ResMLP(size::Symbol = :base; patch_size::Int = 16, imsize::NTuple{2} = (224, 224),
+                drop_path_rate = 0., nclasses = 1000)
+  @assert size in keys(mixer_configs) "`size` must be one of $(keys(mixer_configs))"
+  patch_size = _to_tuple(patch_size)
+  depth = mixer_configs[size][:depth]
+  embedplanes = mixer_configs[size][:planes]
+  layers = mlpmixer(resmixerblock, imsize; mlp_ratio = 4.0, patch_size, embedplanes, 
+                    drop_path_rate, depth, nclasses)
+  ResMLP(layers)
+end
+
+@functor ResMLP
+
+(m::ResMLP)(x) = m.layers(x)
+
+backbone(m::ResMLP) = m.layers[1]
+classifier(m::ResMLP) = m.layers[2]
+
+"""
+    SpatialGatingUnit(norm, proj)
+
+Creates a spatial gating unit as described in the gMLP paper.
+([reference](https://arxiv.org/abs/2105.08050))
+    
+# Arguments
+- `norm`: the normalisation layer to use
+- `proj`: the projection layer to use
+"""
+struct SpatialGatingUnit{T, F}
```

Comment:
Should this go in Layers?

Reply:
This is niche enough that it's used only for this model, which is why I left it here for now - if I discover any future models that reuse it, we could always shift it there